src/Common/tests/System/RealFormatterTestsBase.netcoreapp.cs

```diff
@@ -14,29 +14,118 @@ public abstract class RealFormatterTestsBase
         protected abstract string InvariantToStringDouble(double d, string format);
         protected abstract string InvariantToStringSingle(float f, string format);
 
+        [Theory]
+        [InlineData("#,###.00", double.Epsilon, ".00")]
+        [InlineData("#,###.00", double.MaxValue, "179,769,313,486,232,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000,000.00")]
+        [InlineData("#,###.00", Math.E, "2.72")]
+        [InlineData("#,###.00", Math.PI, "3.14")]
+        [InlineData("#,###.00", 0.0, ".00")]
```

Comment:
I'm surprised this doesn't produce a zero before the decimal - that's how this works on NET Framework?

Reply:
Correct, `#` means generate a digit only if one exists and it is significant. So if it is a leading 0, it generates nothing.